README.md

```diff
@@ -15,10 +16,44 @@ Then alter your homeserver configuration, adding to your `modules` configuration
 modules:
   - module: synapse_user_restrictions.UserRestrictionsModule
     config:
-      # Optional: example boolean flag
-      example_option: True
+      # List of rules. Earlier rules have a higher priority than later rules.
+      rules:
+        - match: "@admin.*:example.org"
+          allow: [invite, create_room]
+
+        - match: "@assistant.*:example.org"
+          allow: [invite]
+
+      # If no rules match, then these permissions are allowed.
+      default_allow: []
+
+      # If no rules match, then these permissions are denied.
+      default_deny: [invite, create_room]
 ```
 
+In this example:
+- `@adminalice:example.org` could create rooms and invite users to
+rooms;
+- `@assistantbob:example.org` could invite users to rooms but NOT create rooms;
+and
+- `@plainoldjoe:example.org` could neither create rooms nor invite users.
+
+### Configuration
+
+Rules are applied top-to-bottom, with the first matching rule being used.
+
+A rule matches if the regular expression (written in `match`) matches the user's ID,
+and the permission being sought is either in the `allow` list or the `deny` list.
+
+Valid permissions (as at the time of writing) are:
+
+- `invite`: the user is trying to invite another user to a room
+- `create_room`: the user is trying to create a room
+
+If no rules match, then `default_allow` and `default_deny` are consulted (in that order);
+`default_deny` is useful for only allowing a select few listed user patterns to
+be allowed to use certain features.
```

Comment:
yeah, that's fair. I was very on the fence about it myself (mostly thinking it would be nice for self-documenting config to say 'these are the things we know/intend are allowed by default'), but I think I'll just remove it because it just opens up questions like 'which one wins: allow or deny?' and does seem a bit redundant. (YAML comments will probably do just as well)